server/src/test/java/org/uiautomation/ios/e2e/nativegestures/ScrollTest.java

```diff
@@ -15,44 +15,107 @@
 
 import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
 import org.testng.Assert;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 import org.uiautomation.ios.BaseIOSDriverTest;
 import org.uiautomation.ios.SampleApps;
 import org.uiautomation.ios.UIAModels.ScrollDirection;
+import org.uiautomation.ios.UIAModels.UIAElement;
+import org.uiautomation.ios.UIAModels.UIAStaticText;
+import org.uiautomation.ios.UIAModels.predicate.*;
 import org.uiautomation.ios.client.uiamodels.impl.RemoteIOSDriver;
 import org.uiautomation.ios.client.uiamodels.impl.RemoteUIAScrollView;
 
 
 public class ScrollTest extends BaseIOSDriverTest {
 
+  private static final String verView1 = "Vertical View 1";
+  private static final String verView2 = "Vertical View 2";
+  private static final String verView3 = "Vertical View 3";
+
+  private static final String horView1 = "Horizontal View 1";
+  private static final String horView2 = "Horizontal View 2";
+  private static final String horView3 = "Horizontal View 3";
+
   @BeforeClass
   public void startDriver() {
     driver = new RemoteIOSDriver(getRemoteURL(), SampleApps.ppNQASampleAppCap());
   }
 
+  private void findElement(String name) {
+    Criteria c1 = new TypeCriteria(UIAStaticText.class);
+    Criteria c2 = new NameCriteria(name, MatchingStrategy.starts);
+    Criteria c = new AndCriteria(c1, c2);
+    UIAElement element = driver.findElement(c);
+  }
+
   @Test
   public void testVerticalScrollingDown() {
     Assert.assertTrue(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 1')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 2')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 3')]").isDisplayed());
+
+    // no Vertical View 2
+    try {
+      findElement(verView2);
+      Assert.fail("shouldn't find element" + verView2);
+    } catch (NoSuchElementException e) {
+      // expected
+    }
+
+    // no Vertical View 3
+    try {
+      findElement(verView3);
+      Assert.fail("shouldn't find element" + verView3);
+    } catch (NoSuchElementException e) {
+      // expected
+    }
 
     ((RemoteUIAScrollView) driver.findElementByXPath("//UIAScrollView[contains(@name, 'Vertical Name')]")).scroll(ScrollDirection.DOWN);
 
     waitForElement(driver, By.xpath("//UIAStaticText[contains(@name, 'Vertical View 2')]"), 6);
     Assert.assertTrue(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 2')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 1')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 3')]").isDisplayed());
+
+    // no Vertical View 1
+    try {
+      findElement(verView1);
```

Comment:
:+1: Good information. Thanks. I still see some tests using the way `try-catch` with `Assert.fail()` in project's tests. I will fix with `expectedExceptions` if I see it.  

Back to the way to fix this test, I can't see how can I use `expectedExceptions` here. After each scroll, the test has to check the absence of two views; therefore, we should have two `@test` that have `expectedExceptions` for each scroll, and we can only do that if we can keep the tests in order we design. However, the `dependsOnMethods` parameter will only guarantee that this method will run after the method it depends on, but not right after that method. I don't see how can I use this way here.

The way to fix that by changing `findElement` is more suitable here.